plugins/processors/ifname/ifname.go

```diff
@@ -0,0 +1,328 @@
+package ifname
+
+import (
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/internal/snmp"
+	si "github.com/influxdata/telegraf/plugins/inputs/snmp"
+	"github.com/influxdata/telegraf/plugins/processors"
+	"github.com/influxdata/telegraf/plugins/processors/reverse_dns/parallel"
+)
+
+var sampleConfig = `
+  ## Name of tag holding the interface number
+  # tag = "ifIndex"
+
+  ## Name of output tag where service name will be added
+  # dest = "ifName"
+
+  ## Name of tag of the SNMP agent to request the interface name from
+  # agent = "agent"
```

Comment:
I considered having an agent setting but didn't think it was worth the potential trouble.  The interface index is a required tag and it's only valid with the agent it came from.  If we add in a way for the index to potentially refer to some other agent, people are going to do that and silently get an interface name from the wrong agent.  I thought requiring the agent to come in as a tag would decrease the possibility that an index is ever used on the wrong agent.

Reply:
Can't this be set to `agent_host` as default? Since this is the tag that is being returned from snmp plugin, It would reduce the need to set this value every time.